.github/workflows/build.yml

```diff
@@ -389,11 +389,11 @@ jobs:
     steps:
       - uses: actions/checkout@v4
       - name: Install poetry
-        run: pipx install poetry
+        run: pipx install 'poetry<2.0'
       - name: Set up Python
         uses: actions/setup-python@v5
         with:
-          python-version: "3.9"
+          python-version: "3.11"
```

Comment:
When updating this, it would be nice to also update [this](https://github.com/OpenRailAssociation/osrd/blob/dev/python/railjson_generator/.python-version). That said, I see that [`actions/setup-python` supports reading `.python-version`](https://github.com/marketplace/actions/setup-python#basic-usage), maybe we can leave the version in one place only?